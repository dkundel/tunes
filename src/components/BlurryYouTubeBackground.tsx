import * as React from 'react';

export interface BlurryYouTubeBackgroundProps {
  videoId: string;
}

export default class BlurryYouTubeBackground extends React.Component<
  BlurryYouTubeBackgroundProps,
  undefined
> {
  public render() {
    if (this.props.videoId && this.props.videoId.length > 0) {
      const url = `https://img.youtube.com/vi/${this.props
        .videoId}/default.jpg`;
      return <img className="player-background blur-background" src={url} />;
    }

    return <img className="player-background blur-background" />;
  }
}
