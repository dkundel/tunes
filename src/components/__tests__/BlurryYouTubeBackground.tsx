import * as React from 'react';
import BlurryYouTubeBackground from '../BlurryYouTubeBackground';
import * as renderer from 'react-test-renderer';

describe('test BlurryYouTubeBackground component', () => {
  test('handles videoId correctly', () => {
    const component = renderer.create(
      <BlurryYouTubeBackground videoId="testid">
        Facebook
      </BlurryYouTubeBackground>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('handles empty videoId', () => {
    const component = renderer.create(
      <BlurryYouTubeBackground videoId="">
        Facebook
      </BlurryYouTubeBackground>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
