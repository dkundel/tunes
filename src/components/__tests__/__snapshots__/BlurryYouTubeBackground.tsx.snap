// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test BlurryYouTubeBackground component handles empty videoId 1`] = `
<img
  className="player-background blur-background sc-bdVaJa boBCaA"
/>
`;

exports[`test BlurryYouTubeBackground component handles videoId correctly 1`] = `
<img
  className="player-background blur-background sc-bdVaJa boBCaA"
  src="https://img.youtube.com/vi/testid/default.jpg"
/>
`;
